components/mp2660/mp2660.h

```diff
@@ -0,0 +1,405 @@
+/*
+ * Copyright (c) 2017 Gunar Schorcht <https://github.com/gschorcht>
+ * Copyright (c) 2019 Ruslan V. Uss <unclerus@gmail.com>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of itscontributors
+ *    may be used to endorse or promote products derived from this software without
+ *    specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file mp2660.h
+ * @defgroup mp2660 mp2660
+ * @{
+ *
+ * ESP-IDF driver for Monolithic Systems 5V USB, 500mA, I2C-Controlled Linear Charger with Power Path Management for Single-Cell Li-Ion Battery
+ *
+ * Copyright (c) 2023 Manuel Markwort <https://github.com/mmarkwort>\n
+ *
+ * BSD Licensed as described in the file LICENSE
+ */
+
+#ifndef __MP2660_H__
+#define __MP2660_H__
+
+#include <stdbool.h>
+#include <i2cdev.h>
+#include <esp_err.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define MP2660_I2C_ADDR 0x09 //!< I2C address
+#define MP2660_SOURCE_CONTROL_REG_DEFAULT 0x4F
+#define MP2660_POWER_ON_CONFIG_REG_DEFAULT 0x4
+#define MP2660_CURENT_CONTROL_REG_DEFAULT 0xE
+#define MP2660_PRECHARGE_TERMINATION_CURRENT_REG_DEFAULT 0x4A
+#define MP2660_CHARGE_VOLTAGE_CONTROL_REG_DEFAULT 0xA3
+#define MP2660_CHARGE_TERMINATION_TIMER_CONTROL_REG_DEFAULT 0x4A
+#define MP2660_MISC_OPERATION_CONTROL_REG_DEFAULT 0xB
+#define MP2660_SYSTEM_STATUS_REG_DEFAULT 0x0
+#define MP2660_FAULT_REG_DEFAULT 0x0
+
+// Input Source Control Register/Address: 00h (Default: 0100 1111)
```

Comment:
All enums, structures, unions and their fields must be commented in Doxygen style.
```C
/**
 * Super-duper enum
 */
typedef enum
{
     ENUM_VAL_0,  //!< Important enum value
     ENUM_VAL_1,  //!< Not so important enum value
} my_enum_t;

/**
 * Config structure
 */
typedef struct
{
    int foo;   //!< Comment to field
} my_struct_t;
```

and so on

Reply:
Hoping, I commented now all correctly